kart/commit.py

```diff
@@ -89,12 +144,21 @@ def commit(ctx, message, allow_empty, output_format, filters):
 
     working_copy.assert_db_tree_match(tree)
 
-    repo_key_filter = RepoKeyFilter.build_from_user_patterns(filters)
-    wc_diff = working_copy.diff_to_tree(repo_key_filter)
+    commit_diff_writer = CommitDiffWriter(repo, "HEAD", filters)
+    wc_diff = commit_diff_writer.get_repo_diff()
 
     if not wc_diff and not allow_empty:
         raise NotFound("No changes to commit", exit_code=NO_CHANGES)
 
+    pk_conflicts = commit_diff_writer.spatial_filter_pk_conflicts
+    if not allow_pk_conflicts and pk_conflicts and any(pk_conflicts.values()):
+        commit_diff_writer.write_warnings_footer()
+        raise InvalidOperation(
+            "Aborting commit due to conflicting primary key values - use --allow-pk-conflicts to commit anyway "
+            "(this will overwrite the existing features with these PK values)",
```

Comment:
I'd suggest making mention of the spatial filter here, because otherwise I don't think this will make much sense to a user:

```suggestion
            "Aborting commit due to conflicting primary key values - use --allow-pk-conflicts to commit anyway "
            "(this will overwrite some existing features outside of the current spatial filter)",
```

Reply:
oh, except `write_warnings_footer` is just above this, so that probably provides enough context for the user to figure this out 👍

Reply:
Done anyway